src/lib/signature_lib/schnorr.ml

```diff
@@ -165,15 +211,37 @@ module Make
                   and type curve_var := Curve.var
                   and type field := Impl.Field.t
                   and type field_var := Impl.Field.Var.t
-                  and type ('a, 'b) checked := ('a, 'b) Impl.Checked.t) :
+                  and type ('a, 'b) checked := ('a, 'b) Impl.Checked.t)
+    (Message_for_mainnet : Message_for_mainnet_intf
```

Comment:
Is there value in having all these separate modules (esp. when the caller of the functor knows all the message types are equal)?

I think it would be nicer to expose from the functor a verification function that takes the hash function as an argument, and then just pass in the mainnet and testnet hash functions in top-level definitions of `verify_for_mainnet` and `verify_for_testnet`. Or maybe even just `verify_for : Signature_kind.t -> Message.t -> bool`

Reply:
I tried something like that, and ran into type difficulties, but I'll give it another try.

Reply:
@imeckler I got rid of all that machinery, which appears not to be needed. See the PR introductory comment.

Reply:
My error, I was always using the testnet hash, so testnet signatures always verified.

Fixed now, and I checked that pretending we had a mainnet signature did not verify.